### ISSUE TITLE
CLI( tns ) Only can deploy first time to Real IPhone device, please help :(

### ISSUE THREAD

**Issue by enchev** [NativeScript/nativescript-cli#1398]
_From @wisonye on January 15, 2016 7:22_

When developing, I use the this command to run in Emulator, everything is fine:

``` javascript
tns livesync ios --emulator --device "iPhone 5" --watch
```

But after I want to test on real IPhone device, I run 

``` livesync
tns run ios
```

It ONLY CAN WORKS for the first time, after that, it always DEPLOY FAIL with this error:

``` javascript
Results at '/Users/wison/NativeScriptProject/DemoMobileApp/platforms/ios/build/device/DemoMobileApp.ipa' 
Project successfully built
Using  /Users/wison/NativeScriptProject/DemoMobileApp/platforms/ios/build/device/DemoMobileApp.ipa
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
```

I really have no idea on this, and the weird thing is keep going: After I unplug the IPHONE, the "tns device" command still can LIST my iphone( how could that possible ):

``` javascript
Wisons-MacBook-Pro:DemoMobileApp wison$ tns device
┌───┬─────────────┬──────────┬──────────────────────────────────────────┐
│ # │ Device Name │ Platform │ Device Identifier                        │
│ 1 │ iPhone5,2   │ iOS      │ 4fcae2e62d5bd99d901647c84b78a3259f71f9ed │
└───┴─────────────┴──────────┴──────────────────────────────────────────┘
```

Anybody can help on this ??? :)

_Copied from original issue: NativeScript/NativeScript#1379_

**Comment by Fatme** [comment]
Hi @wisonye,

Can you try to restart your device and run the command again?

**Comment by wisonye** [comment]
@Fatme I found a tips, after restart my MacOSX, then plugin the IPhone, and the default "Photos" app will open( automatic ), IF I DON'T CLOSE Photos app, "tns run ios" works for more times, if I close it, just fail, I just found this, not sure it's the real reason, still confusing on this :)

**Comment by Fatme** [comment]
It's a really strange. Can you check if `Sync with this iPhone over Wi-Fi` option is turned on from your iTunes?

**Comment by alpipego** [comment]
I have a similar problem. Where I get the same error message:

```
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
```

After unplugging the phone `tns device` does still list the phone, it is not visible in Xcode &rarr; Window &rarr; Devices however.

**Comment by alg** [comment]
It's curious, but I got rid of the error by stopping a ton of apps (mostly games) my son started while I was away. Got to the springboard-only state and it started installing everything smoothly.

**Comment by aylesm** [comment]
I frequently get this issue:

Successfully deployed on device 204ea16234def71a7e7813d9e7490e54b0037b73
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239

Nothing mentioned here works a 100% of the time.

**Comment by rleclair65** [comment]
Same problem with iPad4,1. No solution mentioned works. I start my ios app my xcode.

**Comment by wisonye** [comment]
@rleclair65 Yep, you're right, when "tns" sometime happy, sometime is not happy, the finally helper is XCode, 100% can install and run on IOS Device :) And XCode also fixed the "Black launch screen" issue as well :)

**Comment by aylesm** [comment]
I found that in my terminal if I list the processes and kill the forked server process it helps.  For example run:
`ps`
to get the list of processes.  It should give you something like:

PID TTY           TIME CMD
 496 ttys000    0:00.47 -bash
2660 ttys000    0:00.26 adb -P 5037 fork-server server --reply-fd 4

Then kill fork-server like this:
`kill 2660`

**Comment by alg** [comment]
@aylesm How `adb` is related to iOS?

**Comment by spstratis** [comment]
Has there been any further updates on this?

**Comment by enchev** [comment]
Hey @spstratis what is your CLI version?

**Comment by manijak** [comment]
Someone care to explain why this is closed and no permanent solution provided? 

I have this issue on CLI 2.0.1

**Comment by enchev** [comment]
Hey @manijak,

I asked for more info however never got a response. Hey @dtopuzov @vchimev can you confirm this?

**Comment by manijak** [comment]
It's a verry odd issue. I had an ipad in the other room that had sync over wifi enabled. So it appeared in the device list all the time, even after I disabled the sync.
Had to turn off wifi on the device for it to go away, restarted my Mac and now it works...

**Comment by dtopuzov** [comment]
@manijak we had several issues when deploy/run/debug on real device when --device option is specified. Those issues should be resolved in 2.1.0 (it will be available at the end of this month).

If you still experience some issues with 2.1.0, please reopen or report new issue.

**Comment by christocracy** [comment]
I just got this error with `2.1.1`.  Closed iTunes and it worked.

**Comment by guru1966** [comment]
If you get the following error:
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239

and nothing of the above mentioned tricks help, I usually do the following to fix it:
Disconnect your device and test to run : tns device
If you still see connected devices despite of disconnecting, run the following:
tns run ios --emulator  --watch
This will kill any hanging connected device. Interrupt the command line (ctrl-x,ctrl-c) and close the Iphone emulator and connect your device again and  run again:
tnd run iOS --watch

This is very odd issue that still cause me trouble every time I test on my device.

**Comment by avatsaev** [comment]
got this problem on 2.2.1, still can't figure out why

**Comment by avatsaev** [comment]
Got this error on 2.3.0 (released today 15 sep 2016)
can't deploy on real device 

executed command: tns deploy ios --device 1 

here is the log trace: https://gist.github.com/avatsaev/fdb34ca396a47de2ff65e8bfcb6ca4a5

**Comment by ignaciolarranaga** [comment]
Had the same problem with 2.3.0

**Comment by andreasotto** [comment]
I can confirm the problem still exists (for a long time now ;-( ). 
Rebooting the iPhone temporarily helps, as exactly one install on the phone is possible then. 

But this is far from a solution, one can't reboot the phone for every deploy!

```
# tns run ios --device 1
[....]
Project successfully built
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
```

I'm currently in the process from changing from Ionic to NativeScript, but this issue definitively stops me being productive. And knowing that this issue exists since nearly a year makes me scared. If there is anything i could help with, let me (us) know.

**Comment by hvadotnarola** [comment]
`tns run ios --emulator`

This worked for me.

**Comment by xerotolerant** [comment]
Closing Itunes worked once. It hasn't really worked since. I use ipa-deploy on the side but even that is a a little inconvenient. Is there a way to force disconnect devices in the command line?

**Comment by andreasotto** [comment]
Closing iTunes makes no difference. I have to reboot the iPhone everytime i would like to run or deploy the app again. That makes the development cycle terrible. Is there no help in sight? Do we know if the NativeScript people at least work in this issue?

**Comment by xerotolerant** [comment]
Wait, try ejecting you iPhone from iTunes and closing the app on the device
completely. That works for me off and no. Not 100% of the time but often
enough that I try it before i use ipadeploy.

**Comment by andreasotto** [comment]
I just created a completely new app with the create command. That app is not running on the phone and doesn't need to be stopped. I disconnect the phone from iTunes and closed iTunes.

Because i already had run another app before, i run into the error again:

```
...
Project successfully built
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
```

It's everytime the same. The only solution is the completely switch off the phone (reboot), what isn't a "solution" at all. I don't know how to to keep on developing. In bought an android phone to go on with developing, but thats not a solution, too - because now i'm in debugging iOS and i'm rebooting the phone like 20 times a day.

**Comment by avatsaev** [comment]
Okay i still have this issue so I'm tweeting to @nativescript with a link to this thread, they respond faster on twitter than on github

**Comment by rosen-vladimirov** [comment]
Hey guys, 
I'm reopening this issue as it seems there are still people who have problems with:
```
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
```
errors.

I'll need your help in order to investigate and resolve this issue permanently. Can all of you provide the following information, so we'll be able to reproduce the issue and try to fix it:
 * What's the version of your Mac OS X?
 * What's the version of your XCode ?
 * What's the version of iTunes that you have installed? 
 * When you reproduce the issue, do you have more than one connected devices?
 * Do you have iOS Simulator running while you reproduce the issue?
 * What's the output of `tns device` command when you reproduce the issue?
 * Do you reproduce the issue when you do not use the `--device` option?
 * Do you have `Sync with this iPhone over Wi-Fi` option enabled in iTunes for ANY of your devices?
 * When you reproduce the issue - how many devices do you see connected in iTunes?

I'm looking forward for more information on this case.

**Comment by xerotolerant** [comment]
ok I get this error in the cli
`Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239`

I am running OSX 10.11.3
XCODE 7.3.1
iTunes 12.3.2.35
2 devices one real and one simulator. Killed simulator , same result.
When one real device is connected tns device says one device with a name, id type device and status connected

I do have sync this iphone over wifi enabled for the  device
when sync iphone over wifi was disabled tns devices now does not show the device name and status is unreachable for a while. After a few tests it seems like disabling sync over wifi is the solution. It hasn't occured since I disabled it.

**Comment by rosen-vladimirov** [comment]
@xerotolerant thanks for the detailed information.
@avatsaev , @andreasotto , @ignaciolarranaga can you try disabling the "Sync over wifi" option and see if this will work for you. In case it doesn't, can you answer the questions from my previous post and post them here.

**Comment by avatsaev** [comment]
@rosen-vladimirov deployment was very slow but it did work yes.

**Comment by ignaciolarranaga** [comment]
Sorry for the delay @rosen-vladimirov , providing the data:

* What's the version of your Mac OS X?: 10.12
* What's the version of your XCode ?: 8.0 (8A218a)
* What's the version of iTunes that you have installed?: 12.5.1.21
* When you reproduce the issue, do you have more than one connected devices?: No
* Do you have iOS Simulator running while you reproduce the issue?: Not always, but normally yes.
* What's the output of tns device command when you reproduce the issue?
```
tns device
┌───┬───────────────────┬──────────┬──────────────────────────────────────────┬────────┬───────────┐
│ # │ Device Name       │ Platform │ Device Identifier                        │ Type   │ Status    │
│ 1 │ iPhone de Ignacio │ iOS      │ ......12e0cea5c4705195b814e2b2e4aa...... │ Device │ Connected │
└───┴───────────────────┴──────────┴──────────────────────────────────────────┴────────┴───────────┘
```

* Do you reproduce the issue when you do not use the --device option?: Yes
* Do you have Sync with this iPhone over Wi-Fi option enabled in iTunes for ANY of your devices?: Yes
* When you reproduce the issue - how many devices do you see connected in iTunes?: Just 1

**Comment by ignaciolarranaga** [comment]
@rosen-vladimirov I reproduced the issue, then disabled "Sync over wifi" as requested, ran `tns run ios --device 1` again and same result:

```
Project successfully built
Unable to start service com.apple.mobile.installation_proxy. Result code is: 3892314239
# run ios
....
```

**Comment by avatsaev** [comment]
@ignaciolarranaga Can you try with device UUID instead of 1 (the index)

**Comment by rosen-vladimirov** [comment]
As expected, the problem is caused by the "Sync over Wi-Fi" option. I've modified the code the show helpful message how to resolve such problems, when CLI is unable to work with the device. As CLI is using iTunes to connect to iOS devices, I'm not sure we'll be able to provide a real fix for such cases. But the helpful message describes required actions in order to resolve the issue.

**Comment by avatsaev** [comment]
How does ionic do it? They use a different tool to deploy?

**Comment by hvrz39** [comment]
I'm still having the same issue, just run 'tns run ios --device ######' and got the 'NOENT: no such file or directory, scandir '/Users/...' message. None of the mentioned workarounds above worked for me. Any update on this issue?

**Comment by demetrio812** [comment]
Hello, not sure why this is closed, I'm still having the same problem:

OSX 10.12.3
XCODE 8.2.1
iTunes 12.5.5.5
NativeScript 2.5.0

I don't have "Sync over Wifi" (I just use iCloud from the phone) but I confirm that "tns device" keeps returning the device as connected even when I disconnect it from the cable.

In my case even a restart of the phone doesn't do anything, can you please let me know what I can try?

Thanks,
Dem

**Comment by jawainc** [comment]
you need to create a project in xcode and then run it on your iphone. After the error will be gone.

**Comment by demetrio812** [comment]
Thanks @jawainc 
If I open the NativeScript iOS project with XCode, it compiles and runs properly in the phone, but again when I try it from the command line I get the same error...am I doing something wrong?

Any other idea?

Thanks,
Dem

**Comment by jawainc** [comment]
try removing ios platform and adding again. restart your iphone.
make sure phone and xcode are updated

**Comment by rosen-vladimirov** [comment]
Hey @demetrio812 ,
Can you check if `Sync over Wi-fi` is disabled in iTunes on your machine. Also, can you just try disabling your Wi-fi and execute `tns device` (when your device is connected and disconnected) - maybe some other device is connected to the same Wi-fi and this causes some confusion  in the way CLI communicates to devices.

**Comment by demetrio812** [comment]
Thanks a lot @rosen-vladimirov , I didn't have `Sync over Wi-Fi` on but after disabling the wifi on the phone it disappeared from the list of the devices and then I reconnected and was able to run the app...let's see how long it will pass before I will have to do the procedure again...

Thanks!

**Comment by Roar1827** [comment]
I am experiencing this same issue. Some of the suggestions above help temporarily.

**Comment by odubuc** [comment]
Same issue for me also.

OSX 10.12.3
XCODE 8.2.1
iTunes 12.5.5.5
NativeScript 2.5.0

Sync over wifi is off, I have to reboot the phone on every build.. otherwise 

> Unable to kill the application.

**Comment by dtopuzov** [comment]
@Roar1827 @odubuc Can you reproduce the problem with hello-world application and provide exact steps to reproduce?

**Comment by odubuc** [comment]
I tried several times to reproduce using the hello world application but it worked fine every time.

After a few days of success and errors it seems that I'm getting the error "Unable to kill the application" only after a fatal error in the code somewhere. 

For exemple, this morning I'm getting this one triggered in a save function. (I'm probably trying to save something that nativescript-couchbase doesn't like) :
```
CONSOLE LOG file:///app/provider/error.service.js:40:32: NativeScriptError: Error: -[__NSDate cbjson_encodeTo:canonical:]: unrecognized selector sent to instance 0x17020b250

libc++abi.dylib: terminating with uncaught exception of type NSException
```

Now, if I try to run my app again
`tns run ios --device ..........`

`Unable to kill the application`
It'll fail everytime until I reboot the phone

**Comment by odubuc** [comment]
@dtopuzov I just tested the theory, here's how to reproduce using the tap application, all my steps:

```
tns create ngSample --template typescript
cd ngSample
code .
```
edit the package.json for a valid provisionning ID.
edit the onTap function in app/main-view-model.ts add `throw 'I am Groot!';` anywhere inside the function
![i_am_groot](https://cloud.githubusercontent.com/assets/1559169/24299896/fc7d1dd0-1080-11e7-9c9b-7265006b9c89.png)

`tns run ios --device <device_id>`

tap the button, the app should crash and the console stop itself
launch it again
`tns run ios --device <device_id>`

the console will output 
`Unable to kill the application`

**Comment by davek1979** [comment]
I was having the same issue on iOS and found this as a semi-complete solution to the problem (livesync won't work):

1, forget terminal
2, use XCode to BUILD and RUN on your connected device

Device no longer requires restart between builds.
Sign with WIFI is off.
All code changes are reflected in the build (as expected but didn't work for me before as livesync wouldn't work).

**Comment by dtopuzov** [comment]
Hi All,

In {N} 3.0.0 we changed communication with iOS devices and I can not reproduce the issue.

Can you please try {N} 3.0.0 RC and let us know if you still experience the issue.

```
npm uninstall -g nativescript
npm i -g nativescript@rc
tns create MyApp --template tns-template-hello-world@rc
tns platform add ios@rc --path MyApp
tns plugin remove tns-core-modules --path MyApp
tns plugin add tns-core-modules@rc --path MyApp
tns run ios --path MyApp
```

**Comment by dtopuzov** [comment]
@davek1979 @odubuc @demetrio812 @Roar1827 {N} 3.0.0 is now live, so you can test directly with official version.

**Comment by vultix** [comment]
I've just updated to {N} 3.0.0 and am still seeing this issue.

```
> tns run ios

Skipping prepare.
Searching for devices...
Skipping package build. No changes detected on the native side. This will be fast!
Multiple errors were thrown:
Could not start service com.apple.mobile.installation_proxy
```

**Comment by pkoleva** [comment]
Hey, @vultix , 
Would you try to deploy and run the app from XCode on your device and check if it is ok. If it is, then run 
`tns run ios --log trace`
 and paste us the result in the console here, so we can try to find the specific problem.
Thanks!

**Comment by tylerdoll** [comment]
@pkoleva I too am having this issue and it is becoming a real pain to deal with.  I ran the command you mentioned to @vultix. Here is my [log output](https://github.com/NativeScript/nativescript-cli/files/998111/log.txt).

- What's the version of your Mac OS X? **10.11.6**
- What's the version of your XCode ? **8.2.1**
- What's the version of iTunes that you have installed? **12.6**
- When you reproduce the issue, do you have more than one connected devices? **No**
- Do you have iOS Simulator running while you reproduce the issue? **Sometimes.  Usually it happens when I switch from the simulator to my actual device**
- What's the output of tns device command when you reproduce the issue?
**1 │ Tyler's iPhone │ iOS      │ def8a5215aa87d83776f0a7be535b98778cb840e │ Device │ Connected**
- Do you reproduce the issue when you do not use the --device option? **Yes**
- Do you have Sync with this iPhone over Wi-Fi option enabled in iTunes for ANY of your devices? **Tried it both ways and still didn't work**
- When you reproduce the issue - how many devices do you see connected in iTunes? **1**

**Comment by vultix** [comment]
@pkoleva Sorry for taking so long to respond.  Coming back the next day to try to reproduce the error I have been unable to.  I will let you know if it resurfaces.  When it happens it is very difficult to manage.

**Comment by pkoleva** [comment]
Hey, @tylerdoll,

Thank you for the detailed report, but just to make sure we are on the same page - you use nativescript 3.0.0 (3.0.1 is the latest if you haven't tried it you can give it a go) and your steps to reproduce are:

1. You run the app on a simulator with `tns run ios` -> the app is built and deployed successfully
2. Immediately after that you try to run the same app on a paired device with the same command `tns run ios` -> the app skips prepare and build and could not be deployed to the device.

It will be very helpful if you can send us this app's package.json and also if you verify that the same app can be deployed to the device from XCode (in app/platforms/ios there is a xcodeproj file you can use to do so)

Thanks!

**Comment by pkoleva** [comment]
@tylerdoll, I've consulted with our developers and searched the closed issues and it seems like the problem you are facing is actually with iTunes. The solution is to disable Wi-fi Sync in iTunes for this device, judging from the error code it seems to not support it. Another way to try this is to stop the Wi-fi on this device and try again.
Hope one of these helps!

**Comment by toddanglin** [comment]
I can confirm that I also encounter this problem with {N} 3.0.1 and XCode 8.3.2.

The problem started after LiveSync caused the app to encounter an error on launch. After that error, any attempt to re-run the app has failed.

The target iPad does not have WiFi sync in iTunes enabled, and I also disabled Wifi on the device. To resolve, I had to unplug and then reconnect the USB connection to the iPad. After doing that, I was able to redeploy and run the app.

I'll attempt to create an smaller demo project that can recreate this error condition. Hopefully we can find a way to further minimize this edge case error...

**Comment by ahmadsyamim** [comment]
After reboot device, 'Error is: InstallationLookupFailed.'

**Comment by jrz** [comment]
Error is: Could not start service com.apple.mobile.installation_proxy.
Turned wifi off.
Now: Error is: Could not start device session.

Now suddenly it works.

**Comment by speigg** [comment]
I've been encountering this problem on nativescript v3.2.1:
```
Unable to apply changes on device: 4e025c1c924ddc3c69c797d2a82c3ed75adfc4d7. Error is: Could not start device session.
```

**Comment by hosikiti** [comment]
I've also been encountering this problem since the beginning of December while deploying module through XCode9.2 works.

Error message on `tns run ios`
```
Installing...
Unable to apply changes on device: 5f7a1717923abc2ac017b0d0992eadd4b5573a5a. Error is: InstallationLookupFailed.
```

**Comment by hosikiti** [comment]
My problem solved. It seems "Enterprise Distribution" is a cause of this issue. Enterprise app needed to be "trusted" from control panel before use. However, `tns run` command deletes an installed old app every time before installing new one, and "trusted" information is always reset. Because of that `tns run` cannot launch newly installed app.

What I did to fix (just for anyone who is facing same kind of issue):

1. Create iOS development certificate, app id, and provisioning profile.
2. Install the certificate and the provisioning profile to my local machine.
3. Run `tns build ios --provision` to know the provision ID of newly installed provisoning profile.
4. Run `tns device ios` to know my iPhone's device ID.
5. Run `tns run ios --device <device ID> --provision <provision ID>
6. Done!

**Comment by sean-nicholas** [comment]
I had this problem with the simulator. I solved this by checking out my initial commit, running `tns run ios` (issue is gone) and then re-checkout my last commit. Everything worked again. Strange 🤔

**Comment by dtopuzov** [comment]
Wifi devices not longer detected by tns.
Will be released in nativescript@4.1.0

**Comment by stevegardnersa** [comment]
I've skimmed through the above comments and didn't see my solution but sorry if it's a repeat. I found this problem was caused by a non original iPhone cable. As soon as I switched to an original cable the error stopped.

The error I was getting was...

`Unable to apply changes on device: XXXXXXXXXXXXXXXXXX. Error is: Could not start service com.apple.mobile.installation_proxy.`

**Comment by marcosnc08** [comment]
Hi ! I'm having this issue.

`"nativescript": {
    "id": "blah blah blah",
    "tns-ios": {
      "version": "5.0.0"
    },
    "tns-android": {
      "version": "5.3.1"
    }
  },`

The error is

> Error detected during LiveSync on <<device id>> for <<path to app folder>>. Error: InstallationLookupFailed

I disabled wi fi sync for the device on iTunes but nothing changed. 

Any ideas on how to solve this?

Thanks!
